eng/common/scripts/modules/Package-Properties/Package-Properties.psm1

```diff
@@ -0,0 +1,172 @@
+# Helper functions for retireving useful information from azure-sdk-for-* repo
+# Example Use : Import-Module .\eng\common\scripts\modules\Package-Properties\Package-Properties.psm1 -ArgumentList 'JavaScript','C:\Git\azure-sdk-for-js'
+Param
+(
+    [Parameter(Position=0, Mandatory=$true)]
+    [ValidateSet("net","java","js","python")]
+    [string]
+    $Language,
+    [Parameter(Position=1, Mandatory=$true)]
+    [string]
+    $RepoRoot
+)
+
+function Extract-PkgProps ($pkgPath, $pkgName)
+{
+    if ($Language -eq "net") { Extract-DotNetPkgProps -pkgPath $pkgPath -pkgName $pkgName }
+    if ($Language -eq "java") { Extract-JavaPkgProps -pkgPath $pkgPath -pkgName $pkgName }
+    if ($Language -eq "js") { Extract-JsPkgProps -pkgPath $pkgPath -pkgName $pkgName }
+    if ($Language -eq "python") { Extract-PythonPkgProps -pkgPath $pkgPath -pkgName $pkgName }
+}
+
+function Extract-DotNetPkgProps ($pkgPath, $pkgName)
```

Comment:
These functions look like they work against the repo but as @scbedd was hinting at there is a lot that can be shared from extracting the information from a built package as well. So having that support in here would be awesome, of course we can do that as a later step but something to keep in the back of your mind.